This analysis examines the sequence conservations and structural properties of ISGylation sites, which were identified by [Lilliana Radoshevich](https://medicine.uiowa.edu/microbiology/profile/lilliana-radoshevich)

### Requirements:
* [./metaphors\_api](./metaphors_api): Written by [Leszek Pryszcz](https://github.com/lpryszcz). The installation instruction is [here](https://github.com/lpryszcz/metaphors_api). This allows API access to [metaPhOrs database](http://betaorthology.phylomedb.org/).

### Input Files:
* [./data/MS\_ISG15\_Raw.tsv](./data/MS_ISG15_Raw.tsv): List of ISGylation sites based on mass spectrometry data, generated by [Lilliana Radoshevich](https://medicine.uiowa.edu/microbiology/profile/lilliana-radoshevich).
* [./data/clusterinfo.tsv](./data/clusterinfo.tsv): Clustering of protein based on ISGylation status across different infection conditions.
* [./data/ISG15\_taxID.tsv](./data/ISG15_taxID.tsv): Downloaded from [NCBI](https://www.ncbi.nlm.nih.gov/gene/?term=ISG15).
* [./data/taxID\_ISGylation\_class.csv](./data/taxID_ISGylation_class.csv): Describes whether ISGylation is presence in a given species.
* [./data/ASA.table](./data/ASA.table): Residue's solvent accessibility from [Tien et al.](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0080635).

### Protocol
1. [./script/Parse\_MS.py](./script/Parse_MS.py): Retrives ortholog sequences and PDB codes (if any) for individual proteins of interest.
    - Input file:
      - [./data/MS\_ISG15\_Raw.tsv](./data/MS_ISG15_Raw.tsv)
    - Output file:
      - Fasta/Ortholog_*.fa
      - Structure/PDBcodes_*.txt
2. [./script/Parse\_Orthologs.py](./script/Parse_Orthologs.py): Aligns ortholog sequences of individual proteins of interest and computes sequence conservation of sites of interest.
    - Input file:
      - [./data/MS\_ISG15\_Raw.tsv](./data/MS_ISG15_Raw.tsv)
      - Fasta/Ortholog_*.fa
    - Output file:
      - [./result/ISG15_Lys_con.tsv](./result/ISG15_Lys_con.tsv)
      
